test/cache.test.js

```diff
@@ -51,6 +51,19 @@ test('Cache', async (t) => {
         t.equal(err.message, 'fiiii is not defined in the cache')
       })
     })
+
+    test('should bypass storage when ttl is 0', async (t) => {
```

Comment:
I tried to implement a test for this to the best of my ability, but I'm not entirely satisfied with it. I'm not very familiar with `tap`, and I didn't find any examples of mocking in the existing test suite that I could use to ensure that there's no side-effect on the storage engine for queries that should bypass the cache.

I'd appreciate any suggestions on how to improve this.

Reply:
this seems ok to me.

Reply:
Ah no, you should check that the lookup on the storage is never called. Why don't you just override the function?

Reply:
Thanks for the suggestion. I've amended the commit and the test is now checking that .get/.set are never called, instead of relying on the onHit/onMiss callbacks.